diff --git a/avahi-daemon/avahi-daemon.conf b/avahi-daemon/avahi-daemon.conf
index ad69c73..737a3b2 100644
--- a/avahi-daemon/avahi-daemon.conf
+++ b/avahi-daemon/avahi-daemon.conf
@@ -23,9 +23,9 @@
 [server]
 #host-name=foo
 #domain-name=local
-browse-domains=0pointer.de, zeroconf.org
+#browse-domains=0pointer.de, zeroconf.org
 use-ipv4=yes
-use-ipv6=no
+use-ipv6=yes
 #allow-interfaces=eth0
 #deny-interfaces=eth1
 #check-response-ttl=no
