Index: trunk/avahi-daemon/dbus-service-browser.c
===================================================================
--- trunk/avahi-daemon/dbus-service-browser.c (revision 1235)
+++ trunk/avahi-daemon/dbus-service-browser.c (revision 1329)
@@ -106,8 +106,4 @@
     assert(i);
 
-    i_interface = (int32_t) interface;
-    i_protocol = (int32_t) protocol;
-    u_flags = (uint32_t) flags;
-
     m = dbus_message_new_signal(i->path, AVAHI_DBUS_INTERFACE_SERVICE_BROWSER, avahi_dbus_map_browse_signal_name(event));
 
@@ -118,4 +114,8 @@
             flags |= AVAHI_LOOKUP_RESULT_OUR_OWN;
     }
+    
+    i_interface = (int32_t) interface;
+    i_protocol = (int32_t) protocol;
+    u_flags = (uint32_t) flags;
     
     if (event == AVAHI_BROWSER_NEW || event == AVAHI_BROWSER_REMOVE) {

