diff -Naur avahi-0.6.22.vanilla/avahi-gobject.pc.in avahi-0.6.22/avahi-gobject.pc.in
--- avahi-0.6.22.vanilla/avahi-gobject.pc.in	2008-04-26 15:35:06.000000000 +0200
+++ avahi-0.6.22/avahi-gobject.pc.in	2008-04-26 15:35:43.000000000 +0200
@@ -6,6 +6,6 @@
 Name: avahi-gobject
 Description: Avahi Multicast DNS Responder (GLib GObject Support)
 Version: @PACKAGE_VERSION@
-Requires: glib-2.0 gobject-2.0 avahi-glib
+Requires.private: glib-2.0 gobject-2.0 avahi-glib avahi-client
 Libs: -L${libdir} -lavahi-gobject
 Cflags: -D_REENTRANT -I${includedir}
