--- avahi-daemon/static-services.c.old	2006-03-22 01:06:03.487842768 +0100
+++ avahi-daemon/static-services.c	2006-03-22 01:07:21.197029176 +0100
@@ -682,7 +682,7 @@
 
     memset(&globbuf, 0, sizeof(globbuf));
     if (glob(in_chroot ? "/services/*.service" : AVAHI_SERVICE_DIR "/*.service", GLOB_ERR, NULL, &globbuf) != 0)
-        avahi_log_error("Failed to read service directory.");
+        avahi_log_info("Failed to read service directory or no service files in service directory.");
     else {
         for (p = globbuf.gl_pathv; *p; p++)
             load_file(*p);
