--- avahi-autoipd/dhclient-exit-hook.in.old	2007-04-19 13:20:50.000000000 +0200
+++ avahi-autoipd/dhclient-exit-hook.in	2007-04-19 13:21:16.000000000 +0200
@@ -23,11 +23,11 @@
     MEDIUM|ARPCHECK|ARPSEND|NBI)
         ;;
 
-    PREINIT|BOUND|RENEW|REBIND|REBOOT|STOP)
+    PREINIT|BOUND|RENEW|REBIND|REBOOT|STOP|RELEASE)
         # Stopping avahi-autoipd is left for the enter hook
         ;;
     
-    EXPIRE|FAIL|RELEASE|TIMEOUT)
+    EXPIRE|FAIL|TIMEOUT)
         @sbindir@/avahi-autoipd -wD $interface 2> /dev/null
         ;;
 esac
