Index: /trunk/avahi-ui/avahi-ui.c
===================================================================
--- /trunk/avahi-ui/avahi-ui.c (revision 1431)
+++ /trunk/avahi-ui/avahi-ui.c (revision 1448)
@@ -609,4 +609,5 @@
     
     g_return_val_if_fail(d, NULL);
+    g_return_val_if_fail(AUI_IS_SERVICE_DIALOG(d), NULL);
     
     if (d->priv->domain)
@@ -693,5 +694,5 @@
 
     if (d->priv->service_pulse_timeout <= 0)
-        d->priv->service_pulse_timeout = g_timeout_add(100, service_pulse_callback, d->priv);
+        d->priv->service_pulse_timeout = g_timeout_add(100, service_pulse_callback, d);
 
     for (i = 0; d->priv->browse_service_types[i]; i++)
@@ -873,5 +874,5 @@
 }
 
-static void domain_button_clicked(GtkButton *button,  gpointer user_data) {
+static void domain_button_clicked(GtkButton *button, gpointer user_data) {
     GtkWidget *vbox, *vbox2, *scrolled_window;
     GtkTreeSelection *selection;


