diff -Naur avahi-0.6.22.orig/avahi-gobject/Makefile.am avahi-0.6.22/avahi-gobject/Makefile.am
--- avahi-0.6.22.orig/avahi-gobject/Makefile.am	2008-01-03 13:36:25.000000000 +0100
+++ avahi-0.6.22/avahi-gobject/Makefile.am	2008-01-03 13:36:51.000000000 +0100
@@ -63,7 +63,7 @@
 	$(BUILT_SOURCES)
 
 libavahi_gobject_la_CFLAGS = $(AM_CFLAGS) $(GOBJECT_CFLAGS)
-libavahi_gobject_la_LIBADD = $(AM_LDADD) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la ../avahi-glib/libavahi-glib.la $(GOJECT_LIBS)
+libavahi_gobject_la_LIBADD = $(AM_LDADD) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la ../avahi-glib/libavahi-glib.la $(GOBJECT_LIBS)
 libavahi_gobject_la_LDFLAGS = $(AM_LDFLAGS) -export-dynamic -version-info $(LIBAVAHI_GOBJECT_VERSION_INFO)
 
 # correctly clean the generated source files
diff -Naur avahi-0.6.22.orig/avahi-gobject/Makefile.in avahi-0.6.22/avahi-gobject/Makefile.in
--- avahi-0.6.22.orig/avahi-gobject/Makefile.in	2008-01-03 13:36:25.000000000 +0100
+++ avahi-0.6.22/avahi-gobject/Makefile.in	2008-01-03 13:36:51.000000000 +0100
@@ -518,7 +518,7 @@
 @HAVE_GOBJECT_TRUE@	$(BUILT_SOURCES)
 
 @HAVE_GOBJECT_TRUE@libavahi_gobject_la_CFLAGS = $(AM_CFLAGS) $(GOBJECT_CFLAGS)
-@HAVE_GOBJECT_TRUE@libavahi_gobject_la_LIBADD = $(AM_LDADD) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la ../avahi-glib/libavahi-glib.la $(GOJECT_LIBS)
+@HAVE_GOBJECT_TRUE@libavahi_gobject_la_LIBADD = $(AM_LDADD) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la ../avahi-glib/libavahi-glib.la $(GOBJECT_LIBS)
 @HAVE_GOBJECT_TRUE@libavahi_gobject_la_LDFLAGS = $(AM_LDFLAGS) -export-dynamic -version-info $(LIBAVAHI_GOBJECT_VERSION_INFO)
 
 # correctly clean the generated source files
