Index: ga-entry-group.c
===================================================================
--- ga-entry-group.c	(revision 1783)
+++ ga-entry-group.c	(revision 1784)
@@ -556,9 +556,13 @@
 gboolean ga_entry_group_service_remove_key(GaEntryGroupService * service,
                                   const gchar * key, GError ** error) {
     GaEntryGroupServicePrivate *priv = (GaEntryGroupServicePrivate *) service;
+    GaEntryGroupServiceEntry entry;
 
-    g_hash_table_remove(priv->entries, key);
+    entry.value = key;
+    entry.size = strlen(key);
 
+    g_hash_table_remove(priv->entries, &entry);
+
     if (!priv->frozen)
         return ga_entry_group_service_thaw(service, error);
     else
