--- avahi-utils/avahi/Makefile.in.orig	2005-09-05 11:00:49.000000000 +0100
+++ avahi-utils/avahi/Makefile.in	2005-09-05 11:01:22.000000000 +0100
@@ -517,10 +517,10 @@
 	uninstall-avahiPYTHON uninstall-avahiSCRIPTS uninstall-info-am
 
 
-@HAVE_PYGTK_TRUE@@HAVE_PYTHON_DBUS_TRUE@@HAVE_PYTHON_TRUE@ServiceTypeDatabase.py: ServiceTypeDatabase.py.in
-@HAVE_PYGTK_TRUE@@HAVE_PYTHON_DBUS_TRUE@@HAVE_PYTHON_TRUE@	sed -e 's,@PYTHON\@,$(PYTHON),g' \
-@HAVE_PYGTK_TRUE@@HAVE_PYTHON_DBUS_TRUE@@HAVE_PYTHON_TRUE@		-e 's,@pkgdatadir\@,$(pkgdatadir),g' $< > $@
-@HAVE_PYGTK_TRUE@@HAVE_PYTHON_DBUS_TRUE@@HAVE_PYTHON_TRUE@	chmod +x $@
+@HAVE_PYTHON_TRUE@ServiceTypeDatabase.py: ServiceTypeDatabase.py.in
+@HAVE_PYTHON_TRUE@	sed -e 's,@PYTHON\@,$(PYTHON),g' \
+@HAVE_PYTHON_TRUE@		-e 's,@pkgdatadir\@,$(pkgdatadir),g' $< > $@
+@HAVE_PYTHON_TRUE@	chmod +x $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
